### ISSUE TITLE
vcpkg causes recursive `find_package` in `boost`, crashes build

### ISSUE THREAD

**Issue by GavinRay97** [microsoft/vcpkg#27403]
**Describe the bug**

tl;dr = the following error occurs (see entire post for details)

```cmake
CMake Error at build/vcpkg_installed/x64-linux/share/boost/vcpkg-cmake-wrapper.cmake:1 (set):
  Maximum recursion depth of 1000 exceeded
Call Stack (most recent call first):
  build/_deps/vcpkg-src/scripts/buildsystems/vcpkg.cmake:783 (include)
  build/vcpkg_installed/x64-linux/share/boost/vcpkg-cmake-wrapper.cmake:11 (_find_package)
  build/_deps/vcpkg-src/scripts/buildsystems/vcpkg.cmake:783 (include)
  build/vcpkg_installed/x64-linux/share/boost/vcpkg-cmake-wrapper.cmake:11 (_find_package)
 <RECURSIVE...>
```

I have the below`vcpkg.json` and `CMakeLists.txt`:

```json
{
  "$schema": "https://raw.githubusercontent.com/microsoft/vcpkg/master/scripts/vcpkg.schema.json",
  "name": "learning-db",
  "version": "0.0.1",
  "builtin-baseline": "6ca56aeb457f033d344a7106cb3f9f1abf8f4e98",
  "dependencies": ["folly", "catch2", "liburing", "jemalloc", {"name": "fmt", "version>=": "9.0.0"}],
  "default-features": ["seastar"],
  "features": {
    "seastar": {
      "description": "https://github.com/scylladb/seastar",
      "dependencies": [
          {"name": "boost-atomic", "version>=": "1.77.0"},
          {"name": "boost-chrono", "version>=": "1.77.0"},
          {"name": "boost-date-time", "version>=": "1.77.0"},
          {"name": "boost-filesystem", "version>=": "1.77.0"},
          {"name": "boost-program-options", "version>=": "1.77.0"},
          {"name": "boost-system", "version>=": "1.77.0"},
          {"name": "boost-test", "version>=": "1.77.0"},
          {"name": "boost-thread", "version>=": "1.77.0"},
          {"name": "gmp", "version>=": "6.2.1#13"},
          {"name": "numactl", "version>=": "2.0.14"},
          {"name": "hwloc", "version>=": "2.7.1"},
          {"name": "ragel", "version>=": "6.10"},
          {"name": "libgnutls", "version>=": "3.6.15"},
          {"name":  "zlib", "version>=": "1.2.12"},
          {"name":  "yaml-cpp", "version>=": "0.7.0#1"},
          {"name":  "c-ares", "version>=": "1.18.1"},
          {"name":  "cryptopp", "version>=": "8.6.0"},
          {"name":  "lz4", "version>=": "1.9.3#4"},
          {"name":  "fmt", "version>=": "9.0.0"},
          {"name":  "liburing", "version>=": "2.2"}
      ]
    },
    "seastar-dpdk": {
      "description": "DPDK support for Seastar",
        "dependencies": [
            {"name": "dpdk", "version>=": "19.0.5"}
        ]
    }
  }
}
````

```cmake
cmake_minimum_required(VERSION 3.22)
#######################################
# Project Settings
#######################################
set(CMAKE_CXX_STANDARD 23)
set(CMAKE_CXX_STANDARD_REQUIRED ON)
set(CMAKE_VERBOSE_MAKEFILE ON)

include(FetchContent)
FetchContent_Declare(vcpkg
        GIT_REPOSITORY https://github.com/microsoft/vcpkg/
        GIT_TAG 2022.09.27
)
FetchContent_MakeAvailable(vcpkg)
# NOTE: This must be defined before the first project call
set(CMAKE_TOOLCHAIN_FILE "${vcpkg_SOURCE_DIR}/scripts/buildsystems/vcpkg.cmake" CACHE FILEPATH "")

set(VCPKG_TARGET_TRIPLET x64-linux)
set(BOOST_ROOT ${CMAKE_BINARY_DIR}/vcpkg_installed/${VCPKG_TARGET_TRIPLET})
set(BOOST_INCLUDEDIR "${CMAKE_BINARY_DIR}/vcpkg_installed/${VCPKG_TARGET_TRIPLET}/include")
set(Boost_NO_SYSTEM_PATHS ON)
set(Boost_NO_BOOST_CMAKE  ON)

#######################################
# Project External Dependencies
#######################################
FetchContent_Declare(seastar
        GIT_REPOSITORY https://github.com/scylladb/seastar
        GIT_TAG ac59132009f0ebd32a199031ac6579997d147da3
)
FetchContent_MakeAvailable(seastar)

#######################################
# Project
#######################################
project(my_project)

# Required Fedora packages: openssl-devel perl autoconf automake libtool gettext-devel (autopoint) bison lksctp-tools-devel valgrind valgrind-devel
# See "Folly Integration Guide":
# https://github.com/facebook/rocksdb/wiki/RocksDB-Contribution-Guide#install-dependencies
find_package(Catch2 REQUIRED)
find_package(folly REQUIRED)
find_package(Boost REQUIRED)
find_package(seastar CONFIG REQUIRED)

find_package(PkgConfig)
pkg_check_modules(liburing REQUIRED IMPORTED_TARGET GLOBAL liburing>=2.0)
pkg_check_modules(jemalloc REQUIRED IMPORTED_TARGET GLOBAL)

file(GLOB_RECURSE COMMON_SRC_FILES CONFIGURE_DEPENDS src/*.cpp)
add_executable(${PROJECT_NAME} ${COMMON_SRC_FILES})
target_include_directories(${PROJECT_NAME} PUBLIC
        ${CMAKE_CURRENT_SOURCE_DIR}/include
        ${CMAKE_CURRENT_SOURCE_DIR}/src)

target_link_libraries(${PROJECT_NAME}
        PRIVATE
        # Link with latomic for atomic operations (required by folly)
        atomic
        # jemalloc
        PkgConfig::liburing
        PkgConfig::jemalloc
        Catch2::Catch2WithMain
        fmt::fmt
        Folly::folly Folly::folly_deps
        seastar::seastar
)
```

**Environment**

- WSL2 Fedora 38, Kernel 6.0, Clang 15

**Failure logs**
```cmake
-- Running vcpkg install
Detecting compiler hash for triplet x64-linux...
All requested packages are currently installed.
Restored 0 package(s) from /home/user/.cache/vcpkg/archives in 1.496 us. Use --debug to see more details.

Total elapsed time: 6.078 s
catch2 provides CMake targets:

    # this is heuristically generated, and may not be correct
    find_package(Catch2 CONFIG REQUIRED)
    target_link_libraries(main PRIVATE Catch2::Catch2 Catch2::Catch2WithMain)

The package fmt provides CMake targets:

    find_package(fmt CONFIG REQUIRED)
    target_link_libraries(main PRIVATE fmt::fmt)

    # Or use the header-only version
    find_package(fmt CONFIG REQUIRED)
    target_link_libraries(main PRIVATE fmt::fmt-header-only)

The package zlib is compatible with built-in CMake targets:

    find_package(ZLIB REQUIRED)
    target_link_libraries(main PRIVATE ZLIB::ZLIB)

The package boost is compatible with built-in CMake targets:

    find_package(Boost REQUIRED [COMPONENTS <libs>...])
    target_link_libraries(main PRIVATE Boost::boost Boost::<lib1> Boost::<lib2> ...)

folly provides CMake targets:

    # this is heuristically generated, and may not be correct
    find_package(folly CONFIG REQUIRED)
    target_link_libraries(main PRIVATE Folly::folly Folly::folly_deps Folly::follybenchmark Folly::folly_test_util)

The package liburing can be imported via CMake FindPkgConfig module:

    find_package(PkgConfig)
    pkg_check_modules(liburing REQUIRED IMPORTED_TARGET GLOBAL liburing>=2.0)
    
    target_link_libraries(main PRIVATE PkgConfig::liburing)

The package c-ares provides CMake targets:

    find_package(c-ares CONFIG REQUIRED)
    target_link_libraries(main PRIVATE c-ares::cares)

cryptopp provides CMake targets:

    # this is heuristically generated, and may not be correct
    find_package(cryptopp CONFIG REQUIRED)
    target_link_libraries(main PRIVATE cryptopp-static)

The package gmp can be imported via CMake FindPkgConfig module:

    #  gmp
    find_package(PkgConfig REQUIRED)
    pkg_check_modules(gmp REQUIRED IMPORTED_TARGET gmp)
    target_link_libraries(main PkgConfig::gmp)

    # gmpxx
    find_package(PkgConfig REQUIRED)
    pkg_check_modules(gmpxx REQUIRED IMPORTED_TARGET gmpxx)
    target_link_libraries(main PkgConfig::gmpxx)

lz4 provides CMake targets:

    # this is heuristically generated, and may not be correct
    find_package(lz4 CONFIG REQUIRED)
    target_link_libraries(main PRIVATE lz4::lz4)

yaml-cpp provides CMake targets:

    # this is heuristically generated, and may not be correct
    find_package(yaml-cpp CONFIG REQUIRED)
    target_link_libraries(main PRIVATE yaml-cpp)

-- Running vcpkg install - done
-- Found Boost: /home/user/projects/bustub-cpp-reimpl/build/vcpkg_installed/x64-linux/include (found suitable version "1.80.0", minimum required is "1.73.0")  
-- Using vcpkg find package for Threads 
-- Using vcpkg find package for c-ares 1.13;REQUIRED
-- Using vcpkg find package for PkgConfig REQUIRED
-- Using vcpkg find package for cryptopp 5.6.5;REQUIRED
-- Using vcpkg find package for fmt 5.0.0;REQUIRED
-- Using vcpkg find package for lz4 1.7.3;REQUIRED
-- Using vcpkg find package for PkgConfig REQUIRED
-- Using vcpkg find package for Concepts 
-- Using vcpkg find package for GnuTLS 3.3.26;REQUIRED
-- Using vcpkg find package for PkgConfig REQUIRED
-- Using vcpkg find package for LibUring 2.0;REQUIRED
-- Using vcpkg find package for PkgConfig REQUIRED
-- Using vcpkg find package for LinuxMembarrier 
-- Using vcpkg find package for Sanitizers 
-- Could NOT find Sanitizers (missing: Sanitizers_ADDRESS_COMPILER_OPTIONS Sanitizers_UNDEFINED_BEHAVIOR_COMPILER_OPTIONS) 
-- Using vcpkg find package for SourceLocation 
-- Using vcpkg find package for StdAtomic REQUIRED
-- Using vcpkg find package for hwloc 1.11.2;REQUIRED
-- Using vcpkg find package for PkgConfig REQUIRED
-- Using vcpkg find package for lksctp-tools REQUIRED
-- Using vcpkg find package for numactl REQUIRED
-- Using vcpkg find package for rt REQUIRED
-- Using vcpkg find package for yaml-cpp 0.5.1;REQUIRED
-- Using vcpkg find package for PkgConfig REQUIRED
-- Using vcpkg find package for ragel 6.10;REQUIRED
-- Using vcpkg find package for Threads REQUIRED
-- Using vcpkg find package for PthreadSetName REQUIRED
-- Using vcpkg find package for Valgrind REQUIRED
-- Using vcpkg find package for PkgConfig REQUIRED
-- Running vcpkg install
Detecting compiler hash for triplet x64-linux...
All requested packages are currently installed.
Restored 0 package(s) from /home/user/.cache/vcpkg/archives in 1.314 us. Use --debug to see more details.

Total elapsed time: 6.052 s
catch2 provides CMake targets:

    # this is heuristically generated, and may not be correct
    find_package(Catch2 CONFIG REQUIRED)
    target_link_libraries(main PRIVATE Catch2::Catch2 Catch2::Catch2WithMain)

The package fmt provides CMake targets:

    find_package(fmt CONFIG REQUIRED)
    target_link_libraries(main PRIVATE fmt::fmt)

    # Or use the header-only version
    find_package(fmt CONFIG REQUIRED)
    target_link_libraries(main PRIVATE fmt::fmt-header-only)

The package zlib is compatible with built-in CMake targets:

    find_package(ZLIB REQUIRED)
    target_link_libraries(main PRIVATE ZLIB::ZLIB)

The package boost is compatible with built-in CMake targets:

    find_package(Boost REQUIRED [COMPONENTS <libs>...])
    target_link_libraries(main PRIVATE Boost::boost Boost::<lib1> Boost::<lib2> ...)

folly provides CMake targets:

    # this is heuristically generated, and may not be correct
    find_package(folly CONFIG REQUIRED)
    target_link_libraries(main PRIVATE Folly::folly Folly::folly_deps Folly::follybenchmark Folly::folly_test_util)

The package liburing can be imported via CMake FindPkgConfig module:

    find_package(PkgConfig)
    pkg_check_modules(liburing REQUIRED IMPORTED_TARGET GLOBAL liburing>=2.0)
    
    target_link_libraries(main PRIVATE PkgConfig::liburing)

The package c-ares provides CMake targets:

    find_package(c-ares CONFIG REQUIRED)
    target_link_libraries(main PRIVATE c-ares::cares)

cryptopp provides CMake targets:

    # this is heuristically generated, and may not be correct
    find_package(cryptopp CONFIG REQUIRED)
    target_link_libraries(main PRIVATE cryptopp-static)

The package gmp can be imported via CMake FindPkgConfig module:

    #  gmp
    find_package(PkgConfig REQUIRED)
    pkg_check_modules(gmp REQUIRED IMPORTED_TARGET gmp)
    target_link_libraries(main PkgConfig::gmp)

    # gmpxx
    find_package(PkgConfig REQUIRED)
    pkg_check_modules(gmpxx REQUIRED IMPORTED_TARGET gmpxx)
    target_link_libraries(main PkgConfig::gmpxx)

lz4 provides CMake targets:

    # this is heuristically generated, and may not be correct
    find_package(lz4 CONFIG REQUIRED)
    target_link_libraries(main PRIVATE lz4::lz4)

yaml-cpp provides CMake targets:

    # this is heuristically generated, and may not be correct
    find_package(yaml-cpp CONFIG REQUIRED)
    target_link_libraries(main PRIVATE yaml-cpp)

-- Running vcpkg install - done
-- Configuring incomplete, errors occurred!
See also "/home/user/projects/bustub-cpp-reimpl/build/CMakeFiles/CMakeOutput.log".
See also "/home/user/projects/bustub-cpp-reimpl/build/CMakeFiles/CMakeError.log".

CMake Error at build/vcpkg_installed/x64-linux/share/boost/vcpkg-cmake-wrapper.cmake:1 (set):
  Maximum recursion depth of 1000 exceeded
Call Stack (most recent call first):
  build/_deps/vcpkg-src/scripts/buildsystems/vcpkg.cmake:783 (include)
  build/vcpkg_installed/x64-linux/share/boost/vcpkg-cmake-wrapper.cmake:11 (_find_package)
  build/_deps/vcpkg-src/scripts/buildsystems/vcpkg.cmake:783 (include)
  build/vcpkg_installed/x64-linux/share/boost/vcpkg-cmake-wrapper.cmake:11 (_find_package)
 <RECURSIVE...>

-- Configuring incomplete, errors occurred!
See also "/home/user/projects/bustub-cpp-reimpl/build/CMakeFiles/CMakeOutput.log".
See also "/home/user/projects/bustub-cpp-reimpl/build/CMakeFiles/CMakeError.log".
```

**Comment by Neumann-A** [comment]
tl;dr whoever is printing: `-- Using vcpkg find package for PkgConfig REQUIRED` is responsible and causing the recursion.

**Comment by Cheney-W** [comment]
Thanks for posting this issue. Please reopen this issue if this is still a problem for you.
